src/OrchardCore/OrchardCore.Navigation.Core/ServiceCollectionExtensions.cs

```diff
@@ -1,20 +0,0 @@
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.DependencyInjection.Extensions;
-
-namespace OrchardCore.Navigation
-{
-    public static class ServiceCollectionExtensions
-    {
-        /// <summary>
-        /// Adds tenant level services.
-        /// </summary>
-        /// <param name="services"></param>
-        /// <returns></returns>
-        public static IServiceCollection AddNavigation(this IServiceCollection services)
-        {
-            services.TryAddScoped<INavigationManager, NavigationManager>();
```

Comment:
This cause issues with services resolution when using `scope.ServiceProvider.GetServices<INavigationManager>()` so I needed to remove it for now unless we find a better solution because it should not do that.

@jtkech

Reply:
@Skrypt 

If you need to register multiple implementations and be able to resolve them with `GetServices()`.

    services.AddScoped<INavigationManager, NavigationManager1>();
    services.AddScoped<INavigationManager, NavigationManager2>();

To prevent to re-add the same implementation

    services.TryAddEnumerable(ServiceDescriptor.Scoped<INavigationManager, NavigationManager1>());
    services.TryAddEnumerable(ServiceDescriptor.Scoped<INavigationManager, NavigationManager2>());

Idem with a generic extension

    public static IServiceCollection AddNavigation<T>(this IServiceCollection services)
    {
        services.TryAddEnumerable(ServiceDescriptor.Scoped<INavigationManager, T>());
    }

Reply:
What you are suggesting here is that TryAddEnumerable will work better than TryAddScoped in this case inside a ServiceCollectionExtensions method?

Reply:
Yes, it depends on what is needed, `TryAddScoped` prevent to re-register the same `ServiceType` whatever the `ImplementationType`, useful in an extension which may be called multiple time. But when you want to override a service implementation you need to use `AddScoped` (or `Replace`) so that the last registered will be resolved when you inject/get a single service of this `ServiceType`.

Here the `ServiceType` is `INavigationManager`, the `ImplementationType` is `NavigationManager`.

When we need to inject an `IEnumerable<IScopedService>` or use `GetServices`, we should not use `TryAddScoped` to register other implementations, most of the time we just use `AddScoped` and that's okay if we are sure that we use it once per `ImplementationType`, otherwise when resolved e.g. with `Getservices` we would get and use the same implementation multiple times.

Finally to register multiple `ImplementationType` of the same `ServiceType` but also to prevent from registering duplicate `ImplementationType` we can use `TryAddEnumerable `.